crates/wash/src/cli/dev.rs

```diff
@@ -47,6 +47,140 @@ use crate::{
     },
 };
 
+/// Helper function to check if a path should be ignored during file watching
+/// to prevent artifact directories from triggering rebuilds
+///
+/// # Arguments
+/// * `path` - The file path to check
+/// * `canonical_project_root` - The canonicalized project root directory
+/// * `ignore_paths` - Set of canonicalized paths that should be ignored
+fn is_ignored(path: &Path, canonical_project_root: &Path, ignore_paths: &HashSet<PathBuf>) -> bool {
+    // Try to get the canonical path, fall back to the original if canonicalization fails
+    let canonical_path = path.canonicalize().unwrap_or_else(|_| path.to_path_buf());
+
+    // Check if the canonical path starts with any of the ignore paths
+    for ignore_path in ignore_paths {
+        if canonical_path.starts_with(ignore_path) {
+            return true;
+        }
+    }
+
+    // Also check relative paths from canonical project root
+    if let Ok(relative_path) = canonical_path.strip_prefix(canonical_project_root) {
+        // Check if any component of the relative path matches common build directories
+        for component in relative_path.components() {
+            if let Some(component_str) = component.as_os_str().to_str() {
+                match component_str {
+                    "target" |           // Rust builds
+                    "build" |            // General builds  
+                    "dist" |             // Distribution builds
+                    "node_modules" |     // Node.js dependencies
+                    ".wash" |            // Wash artifacts
+                    ".git" |             // Git metadata
+                    ".wasmbuild" => {     // WebAssembly build artifacts
+                        return true;
+                    }
+                    _ => {}
+                }
+            }
+        }
+    }
+
+    false
+}
+
+/// Build a set of paths that should be ignored during file watching
+/// Uses project-type-specific defaults when available
+///
+/// # Arguments
+/// * `canonical_project_root` - The canonicalized project root directory
+/// * `artifact_path` - Optional path to the build artifact
+/// * `project_context` - Detected project type for specific ignore patterns
+///
+/// # Returns
+/// A set of canonicalized paths that should be ignored during file watching
+fn build_ignore_set(
+    canonical_project_root: &Path,
+    artifact_path: Option<&Path>,
+    project_context: &ProjectContext,
+) -> HashSet<PathBuf> {
+    let mut ignore_paths = HashSet::new();
+
+    // Add the artifact path's parent directory if provided
+    if let Some(artifact) = artifact_path {
+        if let Some(artifact_dir) = artifact.parent() {
+            if let Ok(canonical) = artifact_dir.canonicalize() {
+                ignore_paths.insert(canonical);
+            }
+        }
+    }
```

Comment:
Hrm, what happens if we're outputting the artifact directly into the source of the project?